src/main/java/org/folio/innreach/domain/service/impl/PatronInfoServiceImpl.java

```diff
@@ -0,0 +1,141 @@
+package org.folio.innreach.domain.service.impl;
+
+import static java.util.Optional.ofNullable;
+import static org.apache.commons.lang3.StringUtils.equalsAny;
+
+import static org.folio.innreach.external.dto.InnReachResponse.Error.ofMessage;
+
+import java.time.OffsetDateTime;
+import java.util.UUID;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.stereotype.Service;
+import org.springframework.util.Assert;
+
+import org.folio.innreach.client.PatronBlocksClient;
+import org.folio.innreach.client.PatronClient;
+import org.folio.innreach.domain.dto.folio.User;
+import org.folio.innreach.domain.dto.folio.patron.PatronBlock;
+import org.folio.innreach.domain.service.CentralPatronTypeMappingService;
+import org.folio.innreach.domain.service.CentralServerService;
+import org.folio.innreach.domain.service.InnReachTransactionService;
+import org.folio.innreach.domain.service.PatronInfoService;
+import org.folio.innreach.domain.service.PatronTypeMappingService;
+import org.folio.innreach.dto.PatronInfo;
+import org.folio.innreach.dto.PatronInfoResponseDTO;
+import org.folio.innreach.external.dto.PatronInfoResponse;
+import org.folio.innreach.external.mapper.InnReachResponseMapper;
+
+@Log4j2
+@Service
+@RequiredArgsConstructor
+public class PatronInfoServiceImpl implements PatronInfoService {
+
+  private static final String ERROR_REASON = "Unable to verify patron";
+
+  private final UserServiceImpl userService;
+  private final CentralPatronTypeMappingService centralPatronTypeMappingService;
+  private final PatronTypeMappingService patronTypeMappingService;
+  private final CentralServerService centralServerService;
+  private final InnReachTransactionService transactionService;
+  private final PatronClient patronClient;
+  private final PatronBlocksClient patronBlocksClient;
+  private final InnReachResponseMapper mapper;
+
+  @Override
+  public PatronInfoResponseDTO verifyPatron(String centralServerCode, String visiblePatronId,
+                                            String patronAgencyCode, String patronName) {
+    PatronInfoResponse response;
+    try {
+      var centralServerId = centralServerService.getCentralServerByCentralCode(centralServerCode).getId();
```

Comment:
I think this isn't the best method to get centralServer ID, because loads too much data with JOIN FETCH statements. Maybe we could add a separate method, something like "getCentralServerIdByCentralCode", what do you think?